src/Validation/Validator.php

```diff
@@ -653,11 +809,126 @@ public function allowEmpty($field, $when = true, $message = null)
             if ($settings[$fieldName]['message']) {
                 $this->_allowEmptyMessages[$fieldName] = $settings[$fieldName]['message'];
             }
+            if ($settings[$fieldName]['flags'] !== null) {
+                $this->_allowEmptyFlags[$fieldName] = $settings[$fieldName]['flags'];
```

Comment:
If `notEmpty()` is called on a field after being marked as `allowEmptyArray()` (or one of the other types) should the flags information be cleared?

Reply:
I am not sure if we should take care such use cases. Even now if someone calls notEmpty() without the 'message' option after he called allowEmpty() with the 'message' option, it will end up with an unexpected result.